### PR TITLE
Added Sphere Mask Node

### PR DESCRIPTION
**Pull request by alindmanUnity**
spheremask

# Added Sphere Mask Node

## Purpose
This adds sphere mask functionality via one new node. Given a starting coordinate and center point, you can create a sphere mask using the Radius and Hardness parameters. Will function in 2d or 3d space, based on the vector coordinates given in the Coords and Center input.

## Changes

- Adds `Sphere Mask` node to node library.

## Risks

- n/a

### CODE REVIEW

File: com.unity.shadergraph/CHANGELOG.md [hunk 1]
```diff
@@ -28,6 +28,11 @@ You can now change the path of Shader Graphs and Sub Graphs. When you change the
 
 This adds gradient functionality via two new nodes. The Sample Gradient node samples a gradient given a Time parameter. You can define this gradient on the Gradient slot control view. The Gradient Asset node defines a gradient that can be sampled by multiple Sample Gradient nodes using different Time parameters.
 
+### Sphere Mask Node
+
+![](.data/sphereMask.png)
+
+This adds sphere mask functionality via one new node. Given a starting coordinate and center point, you can create a sphere mask using the Radius and Hardness parameters. Will function in 2d or 3d space, based on the vector coordinates given in the Coords and Center input.
```
Comment (Katstodian) [inline com.unity.shadergraph/CHANGELOG.md]: With this node, you can create a sphere mask. Indicate the starting coordinate and center point, and the sphere mask will use these with the **Radius** and **Hardness** parameters. Sphere mask functionality works in both 2D and 3D spaces, and is based on the vector coordinates in the **Coords and Center** input.

**Comment by alindmanUnity** [comment]
@Katstodian updated the changelog

File: com.unity.shadergraph/CHANGELOG.md [hunk 2]
```diff
@@ -28,6 +28,11 @@ You can now change the path of Shader Graphs and Sub Graphs. When you change the
 
 This adds gradient functionality via two new nodes. The Sample Gradient node samples a gradient given a Time parameter. You can define this gradient on the Gradient slot control view. The Gradient Asset node defines a gradient that can be sampled by multiple Sample Gradient nodes using different Time parameters.
 
+### Sphere Mask Node
+
+![](.data/sphereMask.png)
+
+With this node, you can create a sphere mask. Indicate the starting coordinate and center point, and the sphere mask will use these with the Radius and Hardness parameters. Sphere mask functionality works in both 2D and 3D spaces, and is based on the vector coordinates in the Coords and Center input.
```
Comment (Katstodian) [inline com.unity.shadergraph/CHANGELOG.md]: @pbbastian 
Please replace this:
With this node, you can create a sphere mask. Indicate the starting coordinate and center point, and the sphere mask will use these with the Radius and Hardness parameters. Sphere mask functionality works in both 2D and 3D spaces, and is based on the vector coordinates in the Coords and Center input.

With this (There is MD formatting in the text:)
With this node, you can create a sphere mask. Indicate the starting coordinate and center point. The sphere mask uses these with the **Radius** and **Hardness** parameters. Sphere mask functionality works in both 2D and 3D spaces, and is based on the vector coordinates in the **Coords and Center** input.

Comment (alindmanUnity) [reply com.unity.shadergraph/CHANGELOG.md]: sorry I missed it! changelog should have the requested formatting now

File: com.unity.shadergraph/Editor/Data/Nodes/Math/Vector/SphereMaskNode.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        protected override MethodInfo GetFunctionToConvert()
+        {
+            return GetType().GetMethod("SphereMask", BindingFlags.Static | BindingFlags.NonPublic);
+        }
+
+        static string SphereMask(
+            [Slot(0, Binding.None)] DynamicDimensionVector Coords,
+            [Slot(1, Binding.None, 0.5f, 0.5f, 0.5f, 0.5f)] DynamicDimensionVector Center,
+			[Slot(2, Binding.None, 0.1f, 0.1f, 0.1f, 0.1f)] Vector1 Radius,
+			[Slot(3, Binding.None, 0.8f, 0.8f, 0.8f, 0.8f)] Vector1 Hardness,
```
[the remark this answers is not in the document]
Comment (alindmanUnity) [reply com.unity.shadergraph/Editor/Data/Nodes/Math/Vector/SphereMaskNode.cs]: nitpick away - should be fixed now

**Pull request merged**